sql/src/main/java/io/crate/breaker/MapSizeEstimator.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.breaker;
+
+import org.apache.lucene.util.Accountable;
+import org.apache.lucene.util.RamUsageEstimator;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Map;
+
+import static org.apache.lucene.util.RamUsageEstimator.NUM_BYTES_ARRAY_HEADER;
+import static org.apache.lucene.util.RamUsageEstimator.NUM_BYTES_OBJECT_REF;
+import static org.apache.lucene.util.RamUsageEstimator.alignObjectSize;
+import static org.apache.lucene.util.RamUsageEstimator.shallowSizeOf;
+import static org.apache.lucene.util.RamUsageEstimator.shallowSizeOfInstance;
+
+public final class MapSizeEstimator extends SizeEstimator<Map<?, ?>> {
+
+    public static final MapSizeEstimator INSTANCE = new MapSizeEstimator();
+    private static final int MAX_DEPTH = 1;
+    private static final int STRING_SIZE = (int) shallowSizeOfInstance(String.class);
+
+    @Override
+    public long estimateSize(@Nullable Map<?, ?> value) {
+        if (value == null) {
+            return RamUsageEstimator.NUM_BYTES_OBJECT_HEADER;
+        }
+        return sizeOfMap(value, 0, UNKNOWN_DEFAULT_RAM_BYTES_USED);
+    }
+
+    private static long sizeOfMap(Map<?, ?> map, int depth, long defSize) {
```

Comment:
Everything below is copy & pasted from `RamUsageEstimator` - the Lucene version in 4.0 doesn't contain these methods.